include/os/linux/spl/sys/list.h

```diff
@@ -181,6 +181,12 @@ list_prev(list_t *list, void *object)
 	return (NULL);
 }
 
+static inline int
+list_link_not_active(list_node_t *node)
+{
+	return ((node->next == LIST_POISON1) && (node->prev == LIST_POISON2));
+}
```

Comment:
It's a little confusing that this is different from `!list_link_active()`.  I am guessing this is to catch cases where only one of the pointers is POISON?  And I'm also guessing that should never happen (only one of the pointers being POISON).

Instead of adding this function, could we make list_link_active() assert that either both or neither pointer is POISON?  e.g. 
```
EQUIV(node->next == LIST_POISON1, node->prev == LIST_POISON2);
return (node->next != LIST_POISON1);
```

Reply:
I did add this to avoid the logic short circuiting on the && statement in link_list_active(). We were observing, due to the race condition, that next would be pointing at a valid ABD, but prev was pointing at LIST_POISON2. I am fine with just updating list_link_active() to the EQUIV statement and return.